### PR TITLE
Remove index management, always use data streams

### LINKED ISSUE #5961: Remove Elasticsearch setup, default to waiting for data streams to be set up
Follow up on https://github.com/elastic/apm-server/issues/5914

In `7.x` an option is introduced to wait for data streams to be set up before writing to data streams. The option is disabled by default. When this option is enabled, APM Server waits for the integration package to be installed before writing any data.

In `8.0` we will 
* remove the ability to setup Elasticsearch
* always write to data streams (remove option)
* turn the option to wait for data streams to be set up on by default. It should still be possible to disable the option for scenarios where APM Server lacks privileges to check the package is installed; in this case the user is responsible for ensuring the package is installed before APM Server is started.

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

- Remove `apm-server.ilm.*` config
- Remove `apm-server.data_streams.enabled` config
- Remove `setup.*` config
- Remove `output.elasticsearch.{index,indices}` config
- Stop building, including, and using libbeat fields

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~ Doc updates being handled separately

## How to test these changes

1. Run apm-server without any special config, other than Elasticsearch host & credentials
2. Install the APM integration package via Kibana
3. Send events to apm-server
4. Check events are indexed into data streams

## Related issues

Closes #5961

### CODE REVIEW

**Review by simitt — COMMENTED** [review]
Are you going to remove the `apm-server.register.ingest.pipeline` in a separate PR?

**Comment by axw** [comment]
> Are you going to remove the apm-server.register.ingest.pipeline in a separate PR?

@simitt do you mean remove from apm-server.yml? It was removed in https://github.com/elastic/apm-server/pull/6575:

https://github.com/elastic/apm-server/commit/642aaee1c2b87618770a412aa99ccda57a237311#diff-fbadce52f216b92bda4dab62b076f334502b170af5e51bf3e442e2033b4a4e9d

**Comment by simitt** [comment]
Yes. Sorry I missed that and still had it in my main branch when I checked.

File: README.md [hunk 1]
```diff
@@ -64,47 +64,14 @@ To run APM Server with debugging output enabled, run:
 
 For Testing check out the [testing guide](TESTING.md)
 
-### Update
```
[the remark this answers is not in the document]
Comment (axw) [reply README.md:67]: Added a brief paragraph, let me know what you think.

**Review by simitt — APPROVED** [review]

**Comment by marclop** [comment]
Verified using 9bb5e4cdd.

**Pull request merged**